adoc/chapters/device_compiler.adoc

```diff
@@ -377,6 +386,14 @@ in <<sec:backends>> must be defined by all conformant implementations.
 
 A number of kernel features defined by this SYCL specification are optional;
 they may be supported on some devices but not on other devices.
+
+As stated in <<sec:language.restrictions.kernels>>, the restrictions for
+optional kernel features do not apply to manifestly constant-evaluated
+expressions or conversions in device code.
+Device code may use optional features in manifestly constant-evaluated
+expressions or conversions even if the device does not support the optional
+feature.
+
 As described in <<sec:device-aspects>>, an application can test whether a device
 supports these features by testing whether the device has an associated aspect.
```

Comment:
Thanks for making the text normative, but I have a new comment now that the location of this paragraph has changed.  The phrase "these features" refers to the optional features described in the first paragraph.  However, your new paragraph intervenes, making it less clear what "these" refers to.  Let's change it to "an optional feature" to clarify what we are talking about:

```suggestion
supports an optional feature by testing whether the device has an associated aspect.
```